app/uk/gov/hmrc/eoricommoncomponent/frontend/config/InternalAuthTokenInitialiser.scala

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.eoricommoncomponent.frontend.config
+
+import play.api.Logging
+import play.api.Configuration
+import play.api.libs.json.Json
+import uk.gov.hmrc.http.{HeaderCarrier, StringContextOps}
+import uk.gov.hmrc.http.client.HttpClientV2
+import uk.gov.hmrc.http.HttpReads.Implicits.readRaw
+import uk.gov.hmrc.eoricommoncomponent.frontend.models.Done
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.duration.DurationInt
+import play.api.http.HeaderNames.AUTHORIZATION
+import scala.concurrent.{Await, ExecutionContext, Future}
+
+abstract class InternalAuthTokenInitialiser {
+  val initialised: Future[Done]
+}
+
+@Singleton
+class NoOpInternalAuthTokenInitialiser @Inject() () extends InternalAuthTokenInitialiser with Logging {
+
+  override val initialised: Future[Done] = {
+    logger.info("Internal auth not active")
+    Future.successful(Done)
+  }
+
+}
+
+@Singleton
+class InternalAuthTokenInitialiserImpl @Inject() (configuration: Configuration, httpClient: HttpClientV2)(implicit
+  ec: ExecutionContext
+) extends InternalAuthTokenInitialiser with Logging {
+
+  private val internalAuthService: Service =
+    configuration.get[Service]("microservice.services.internal-auth")
+
+  private val authToken: String =
+    configuration.get[String]("internal-auth.token")
+
+  private val appName: String =
+    configuration.get[String]("appName")
+
+  override val initialised: Future[Done] =
+    ensureAuthToken()
+
+  Await.result(initialised, 30.seconds)
+
+  private def ensureAuthToken(): Future[Done] =
+    authTokenIsValid.flatMap { isValid =>
+      if (isValid) {
+        logger.info("Auth token is already valid")
+        Future.successful(Done)
+      } else
+        createClientAuthToken()
+    }
+
+  private def createClientAuthToken(): Future[Done] = {
```

Comment:
@poscala so this initialiser only runs when it's configured to - and we'll only configure it to run locally/via jenkins.

This is just because internal-auth doesn't check for real config locally, so we need to use a test endpoint of their service to set up our tokens. This won't run in QA/Staging/Production.

I don't think it'll need to be refreshed at all unless internal-auth gets restarted.